controllers/selfnoderemediation_controller.go

```diff
@@ -157,8 +157,12 @@ func (r *SelfNodeRemediationReconciler) Reconcile(ctx context.Context, req ctrl.
 	defer func() {
 		if updateErr := r.updateSnrStatus(ctx, snr); updateErr != nil {
 			if apiErrors.IsConflict(updateErr) {
-				if returnErr == nil && !returnResult.Requeue {
-					returnResult = ctrl.Result{RequeueAfter: time.Second}
+				minRequeue := time.Second
+				if returnResult.Requeue && minRequeue > returnResult.RequeueAfter {
```

Comment:
I wonder if the condition on `returnResult.Requeue` is necessary. Could it just be

```
if minRequeue > returnResult.RequeueAfter {
    minRequeue = returnResult.RequeueAfter
}
```

?

Reply:
Default value for `RequeueAfter` is 0 when it isn't set  :man_shrugging:

Reply:
I don't understand the logic implemented here tbh.

- The idea was to always requeueu quickly for conflicts, not?
- The `if` statement here does not make sense,  because `Requeue` can be false while `RequeuAfter` is set (will still trigger a requeue)
- The check for a `RequeueAfter` < 1s makes this too verbose for no value IMHO. Even when some of the calculations we do result in < 1s, IMHO 1s is always quick enough. Even more knowing that this is running potentially on many many nodes... let's not stress the API server too much :) 

IMHO this should be enough:
```
if updateErr := r.updateSnrStatus(ctx, snr); updateErr != nil {
    if apiErrors.IsConflict(updateErr) {			
        returnResult.RequeueAfter = 1 * time.Second
    else {
        returnErr = utilerrors.NewAggregate([]error{updateErr, returnErr}) // handles nil errors nicely
    }
}
```

Reply:
>The idea was to always requeueu quickly for conflicts, not?

Yes, but in case the requeque of the original result is lower than that of the conflict I'd prefer not to override it.

>The if statement here does not make sense, because Requeue can be false while RequeuAfter is set (will still trigger a requeue)

That's interesting,  initially reading the comment on `RequeueAfter` I've assumed this not the case (i.e Requeue is set to true), after your comment and on a second read I'm not so sure.

I think that in order to stay on the safe side I'll replace `Requeue` with `RequeueAfter > 0` that should assure that the lowest requeue time will be applied.